src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -406,13 +392,10 @@ private void emitMetrics(long currWindowStartTime) throws Exception {
 
     metricsDB.commit();
     metricsDBMap.put(prevWindowStartTime, metricsDB);
-    long metricsDBSize = new File(metricsDB.getDBFilePath()).length();
     PerformanceAnalyzerApp.READER_METRICS_AGGREGATOR.updateStat(
-        ReaderMetrics.METRICSDB_FILE_SIZE, "", metricsDBSize);
+        ReaderMetrics.METRICSDB_FILE_SIZE, "", new File(metricsDB.getDBFilePath()).length());
```

Comment:
Instead of creating a new file and getting its length, can we replace `new File(metricsDB.getDBFilePath()).length()` with `metricsDB.getDBFilePath().length()`?

Reply:
I don't think that would work. `getDBFilePath` actually returns a String, not a Path. Also, a Path doesn't have a method to retrieve the size of the underlying file.